src/string.cr

```diff
@@ -4428,6 +4428,134 @@ class String
       value
     end
   end
+
+  # Implementation of string interpolation of a single string.
+  #
+  # For example, this code will end up invoking this method:
+  #
+  # ```
+  # value = "hello"
+  # "#{value}" # same as String.interpolation(value)
+  # ```
+  #
+  # In this case the implementation just returns the same string value.
+  #
+  # NOTE: there should never be a need to call this method instead of using string interpolation.
+  def self.interpolation(value : String)
```

Comment:
What's the purpose of this overload? It's equivalent to the non-restricted overload. `String#to_s` returns `self`.

Reply:
You are right